### DISCUSSION TITLE
k0s Kubernetes LXC installed but has issue with configuration

### DISCUSSION THREAD

**Discussion by AAH134** [tteck/Proxmox#1320] [Help Needed]
After installing k0s script everything look fine but kubectl commands have some issue related to config file i guess,

If there is any special things have to be done after installing how can I get going? 

I was playing around with my homelab to get a kubernete setup but have no success.

This proxmox have latest update, running on i7 3770 with 32GB ddr3 dell opteplex 

Is it a bug with me? Or I have to modify config file?

**Comment by liveware-problems** [comment]
Personally, I couldn't get k0s to start up properly after installing using the script. I did mention the issues in https://github.com/tteck/Proxmox/discussions/877

I ended up running the LXC install script for Docker, and then I was able to pull and install k0s using the default online guide and it started up without issue. I also created a new Docker LXC and installed k3s and decided to leave that one running.

**Comment by beautyfree** [comment]
`k0s kubectl get nodes`
returns `No resources found`

Can it be the reason? https://stackoverflow.com/questions/75047809/installing-k0s-using-getting-started-guide-shows-no-nodes#comment132437910_75048209

**Reply by tteck** [reply]
https://github.com/tteck/Proxmox/blob/62784e9200cbd0a02c0fb9b485e4f50caa165f74/install/k0s-install.sh#L25

Are you suggesting to remove the flag `--single`?

**Reply by tteck** [reply]
Same results without the `--single`

**Reply by beautyfree** [reply]
> https://github.com/tteck/Proxmox/blob/62784e9200cbd0a02c0fb9b485e4f50caa165f74/install/k0s-install.sh#L25
> 
> Are you suggesting to remove the flag `--single`?



actually no, I would like --single to work, because I want to raise a stand for development on one VM/LXC

**Reply by beautyfree** [reply]
It seems that I got the working solution.

I used k0s LXC script but with Ubuntu 22.04 and privileged container.
And added `/etc/rc.local` in the container like [this](https://gist.github.com/triangletodd/02f595cd4c0dc9aac5f7763ca2264185#etcrclocal)

Also, I added configurations from this [section](https://gist.github.com/triangletodd/02f595cd4c0dc9aac5f7763ca2264185#on-the-host) 
but I don't know how it affected.

And now I see:
```bash
root@k0s:~# sudo k0s kubectl get nodes   
NAME   STATUS   ROLES           AGE     VERSION
k0s    Ready    control-plane   5m50s   v1.27.6+k0s
```
